api_tests/preprints/views/test_preprint_detail.py

```diff
@@ -906,9 +910,8 @@ def test_update_why_no_data(self, app, user, preprint, url):
 
         res = app.patch_json_api(url, update_payload, auth=user.auth, expect_errors=True)
 
-        assert res.status_code == 400
-        assert res.json['errors'][0]['detail'] == 'You cannot edit this statement while your data links availability' \
-                                                  ' is set to true or is unanswered.'
+        assert res.status_code == 200
+        assert res.json['data']['attributes']['why_no_data'] == 'My dog ate it.'
 
         preprint.has_data_links = 'no'
         preprint.save()
```

Comment:
Again, I think it's better for our purposes to have the API requests be complete requests for changing the state, so in this case it would have the proper has-or-doesn't-have-data information in the api request. Consider that a general comment for these api requests.